engine/src/main/scala/cromwell/webservice/LabelsManagerActor.scala

```diff
@@ -50,19 +46,49 @@ class LabelsManagerActor(serviceRegistryActor: ActorRef) extends Actor with Acto
 
   implicit val ec = context.dispatcher
 
+  // These var Options get set by the effective entry point of the actor, the LabelsAddition case
   private var wfId: Option[WorkflowId] = None
+  private var newLabels: Option[Labels] = None
+
   private var target: ActorRef = ActorRef.noSender
 
-  def receive = {
+  override def receive = {
     case LabelsAddition(data) =>
       wfId = Option(data.workflowId)
+      newLabels = Option(data.labels)
       target = sender()
       serviceRegistryActor ! PutMetadataActionAndRespond(labelsToMetadataEvents(data.labels, data.workflowId), self)
-    case MetadataWriteSuccess(events) =>
-      target ! BuiltLabelsManagerResponse(processLabelsResponse(wfId.get, metadataEventsToLabels(events)))
+    case MetadataWriteSuccess(_) =>
+      /*
+        Ask the metadata store for the current set of labels, so we can return the full label set to the user.
+        At this point in the actor lifecycle, wfId has already been filled out so the .get is safe
+      */
+      serviceRegistryActor ! GetLabels(wfId.get)
+    case LabelLookupResponse(id, origLabels) =>
+      /*
+        There's some trickery going on here. We've updated the labels in the metadata store but almost certainly when
+        the store received the GetLabels request above the summarizer will not have been run so our new values are
+        not present. Instead we'll fake it by manually merging what the metadata store has, as well as the new updated
+        values.
+
+        There's a potential flaw here in that a second PATCH request to the labels endpoint could come in prior to
+        summarization and thus the response to the user would lose the previous updates, but we can cross that bridge
+        if it becomes a problem. So if you find yourself debugging why previously updated labels aren't showing up in
+        the return packet, this is a likely cause.
```

Comment:
Thanks for the docstring here, they would be really helpful if we run into this kind of issues.